src/rpc/blockchain.cpp

```diff
@@ -748,6 +750,52 @@ static RPCHelpMan getmempoolentry()
     };
 }
 
+static RPCHelpMan getblockfrompeer()
+{
+    return RPCHelpMan{"getblockfrompeer",
+                "\nAttempt to fetch block from a given peer.\n"
+                "\nWe must have the header for this block, e.g. using submitheader.\n"
+                "\nReturns {} if a block-request was successfully scheduled\n",
```

Comment:
Even that is not guaranteed at the moment (albeit very likely to hold in practice).

Combined with #19463, the caller could set a prune lock to ensure it doesn't get re-pruned before the caller is done with it.

Reply:
So far I wasn't able to figure out in which cases this could not be guaranteed. I opened a follow-up PR in #23813. Please let me know if you have suggestions on improving the text there. Sure #19463 can be helpful but my aim with this for now is to just make sure that users don't get wrong expectations, like that the block would be available forever for example.